src/Symfony/Component/OptionsResolver/OptionConfigurator.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\OptionsResolver;
+
+use Symfony\Component\OptionsResolver\Exception\AccessException;
+
+final class OptionConfigurator
+{
+    private $name;
+    private $resolver;
+
+    public function __construct(string $name, OptionsResolver $resolver)
+    {
+        $this->name = $name;
+        $this->resolver = $resolver;
+        $this->resolver->setDefined($name);
+    }
+
+    /**
+     * Adds allowed values for this option.
+     *
+     * @param mixed ...$allowedValues One or more acceptable values/closures
+     *
+     * @return $this
+     *
+     * @throws AccessException If called from a lazy option or normalizer
+     */
+    public function addAllowedValues(...$allowedValues): self
+    {
+        $this->resolver->addAllowedValues($this->name, $allowedValues);
+
+        return $this;
+    }
+
+    /**
+     * Sets allowed types for this option.
+     *
+     * @param string ...$allowedTypes One or more accepted types
+     *
+     * @return $this
+     *
+     * @throws AccessException If called from a lazy option or normalizer
+     */
+    public function addAllowedTypes(string ...$allowedTypes): self
+    {
+        $this->resolver->addAllowedTypes($this->name, $allowedTypes);
+
+        return $this;
+    }
+
+    /**
+     * Adds a normalizer for this option.
+     *
+     * @return $this
+     *
+     * @throws AccessException If called from a lazy option or normalizer
+     */
+    public function addNormalizer(\Closure $normalizer): self
+    {
+        $this->resolver->addNormalizer($this->name, $normalizer);
+
+        return $this;
+    }
+
+    /**
+     * Prepends a normalizer for this option.
+     *
+     * @return $this
+     *
+     * @throws AccessException If called from a lazy option or normalizer
+     */
+    public function prependNormalizer(\Closure $normalizer): self
```

Comment:
all adder methods and also this one should be removed now that re-defining existent options are no longer allowed.

Reply:
So the only methods that are going to remains are `allowedTypes`, `allowedValues`, `default`, `define`, `deprecated`, `normalize` and `required`. Right?

Reply:
Yes, exactly.